src/storage/authorization/oauth2.cpp

```diff
@@ -0,0 +1,168 @@
+                                       curl_slist *extra_headers) {
+	return APIUtils::GetRequest(context, endpoint_builder, token, extra_headers);
+}
+
+unique_ptr<OAuth2Authorization> OAuth2Authorization::FromAttachOptions(ClientContext &context,
+                                                                       IcebergAttachOptions &input) {
+	auto result = make_uniq<OAuth2Authorization>();
+
+	unordered_map<string, Value> remaining_options;
+	string secret;
+	for (auto &entry : input.options) {
+		auto lower_name = StringUtil::Lower(entry.first);
+		if (lower_name == "secret") {
+			secret = entry.second.ToString();
+		} else if (lower_name == "oauth2_scope") {
+			result->scope = entry.second.ToString();
+		} else if (lower_name == "oauth2_server_uri") {
+			result->uri = entry.second.ToString();
+		} else if (lower_name == "client_id") {
+			result->client_id = entry.second.ToString();
+		} else if (lower_name == "client_secret") {
+			result->client_secret = entry.second.ToString();
+		} else if (lower_name == "oauth2_grant_type") {
+			result->grant_type = entry.second.ToString();
+		} else {
+			remaining_options.emplace(std::move(entry));
+		}
+	}
+
+	Value token;
+	// Check if any of the options are given that indicate intent to provide options inline, rather than use a secret
+	bool user_intends_to_use_secret = true;
+	if (!result->scope.empty() || !result->uri.empty() || !result->client_id.empty() ||
+	    !result->client_secret.empty()) {
+		user_intends_to_use_secret = false;
+	}
+
+	auto iceberg_secret = IRCatalog::GetIcebergSecret(context, secret, user_intends_to_use_secret);
+	if (iceberg_secret) {
+		//! The catalog secret (iceberg secret) will already have acquired a token, these additional settings in the
+		//! attach options will not be used. Better to explicitly throw than to just ignore the options and cause
+		//! confusion for the user.
+		if (!result->scope.empty()) {
+			throw InvalidConfigurationException(
+			    "Both an 'oauth2_scope' and a 'secret' are provided, these are mutually exclusive.");
+		}
+		if (!result->uri.empty()) {
+			throw InvalidConfigurationException("Both an 'oauth2_server_uri' and a 'secret' are "
+			                                    "provided, these are mutually exclusive.");
+		}
+		if (!result->client_id.empty()) {
+			throw InvalidConfigurationException(
+			    "Please provide either a 'client_id'+'client_secret' pair, or 'secret', "
+			    "these options are mutually exclusive");
+		}
+		if (!result->client_secret.empty()) {
+			throw InvalidConfigurationException("Please provide either a client_id+client_secret pair, or 'secret', "
+			                                    "these options are mutually exclusive");
+		}
+
+		auto &kv_iceberg_secret = dynamic_cast<const KeyValueSecret &>(*iceberg_secret->secret);
+		token = kv_iceberg_secret.TryGetValue("token");
+		result->uri = kv_iceberg_secret.TryGetValue("token").ToString();
```

Comment:
this doesn't look right to me

Reply:
Also if you add `input.endpoint = kv_iceberg_secret.TryGetValue("endpoint").ToString();` under this, and move [this check](https://github.com/duckdb/duckdb-iceberg/pull/142/files#diff-629ad72ca849b0e01b7afe5a855c65677fb5e27a1423b0c58abdd5f3f20a8793R178) to after parsing the oauth workflow, we can support (1) from [this comment](https://github.com/duckdb/duckdb-iceberg/pull/142#issuecomment-2794915560)

Reply:
We can also add a log statement here to say we are assuming the endpoint from iceberg secret named 'xyz'